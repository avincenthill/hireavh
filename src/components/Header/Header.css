.header-container {
  color: var(--c_2);
  background-color: var(--c_4);
  position: absolute;
  top: 0;
  width: 100%;
  height: var(--s-s);
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: var(--fs-xs);
}