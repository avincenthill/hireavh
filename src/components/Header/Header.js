import './Header.css';
import { Nav } from 'components/Nav/Nav';
import React from 'react';
import ToggleSwitch from 'components/ToggleSwitch/ToggleSwitch';
import content from 'content/content';
import utils from 'utils/utils';

class Header extends React.Component {
  render() {
    return (
      <div>
        <Nav isBottom={false}></Nav>
        <a href="/about" className="header-container">
          <div className="header-switch-theme">
            <ToggleSwitch
              id="toggle-switch-theme"
              onChange={() => {
                utils.toggleTheme();
              }}
              defaultChecked={utils.isDarkTheme()}
              text={content.header.switchOptions}
            ></ToggleSwitch>
          </div>
          {content.header.title}
        </a>
      </div>
    );
  }
}

export default Header;
