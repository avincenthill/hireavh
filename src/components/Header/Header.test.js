import React from "react";
// import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import Header from "./Header";

describe("Header", () => {
  describe("snapshot", () => {
    it("matches the last saved snapshot", () => {
      const tree = renderer.create(<Header />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
