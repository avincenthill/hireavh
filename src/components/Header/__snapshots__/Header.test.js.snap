// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header snapshot matches the last saved snapshot 1`] = `
<div>
  <div
    className="nav nav-top"
  >
    <div
      className="nav-icon $
          this.props.isBottom ? \\"nav-icon-bottom\\" : \\"nav-icon-top\\"
        }"
    >
      <svg
        fill="currentColor"
        height="1em"
        onClick={[Function]}
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": "var(--c-1)",
            "height": "3rem",
            "width": "3rem",
          }
        }
        viewBox="0 0 8 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8 4v1H0V4h8zM0 8h8V7H0v1zm0 3h8v-1H0v1z"
          fillRule="evenodd"
        />
      </svg>
    </div>
  </div>
  <a
    className="header-container"
    href="/about"
  >
    <div
      className="header-switch-theme"
    >
      <div
        className="toggle-switch"
      >
        <input
          className="toggle-switch-checkbox"
          defaultChecked={true}
          id="toggle-switch-theme"
          onChange={[Function]}
          type="checkbox"
        />
        <label
          className="toggle-switch-label"
          htmlFor="toggle-switch-theme"
        >
          <span
            className="toggle-switch-inner"
            data-no="üåô"
            data-yes="‚òÄÔ∏è"
          />
        </label>
      </div>
    </div>
    üè† about
  </a>
</div>
`;
